pmd-apex/src/test/resources/net/sourceforge/pmd/lang/apex/rule/performance/xml/EagerlyLoadedDescribeSObjectResult.xml

```diff
@@ -89,4 +89,40 @@ public class Foo {
         ]]></code>
     </test-code>
 
+    <test-code>
+        <description>No describer options on SObjectField</description>
+        <expected-problems>1</expected-problems>
+        <expected-linenumbers>3</expected-linenumbers>
+        <code><![CDATA[
+public class Foo {
+    public void bar(Case case) {
+        String fieldName = Case.Subject.getDescribe().getName();
+    }
+}
+        ]]></code>
+    </test-code>
```

Comment:
@rsoesemann Can you have a look at this?

Should the rule [EagerlyLoadedDescribeSObjectResult](https://pmd.github.io/latest/pmd_rules_apex_performance.html#eagerlyloadeddescribesobjectresult) actually detect SObjectField.getDescribe calls? Or is this a false positive?

As far as I understand from the documentation on [SObjectField](https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_class_Schema_SObjectField.htm), there is no such performance issue as in `SObjectType.getDescribe()`. Although the doc is very vague in what exactly is loaded ("Returns the describe field result for this field."). However, with `FieldDescribeOptions.FULL_DESCRIBE` one can get all informations. But this doesn't seem to be not the default behaviour and therefore is not a performance issue. There is a DEFAULT constant however as well ([FieldDescribeOptions Enum](https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_enum_Schema_FieldDescribeOptions.htm)), which says "Compute context-specific, describe field results.".
That's why I think, the rule shouldn't flag this case.

What do you think?

Reply:
I've decided now to declare this a false positive